yb-voyager/cmd/exportData_test.go

```diff
@@ -1081,3 +1082,270 @@ func TestTableListInFreshRunOfExportDataBasicYB(t *testing.T) {
+	err = InitNameRegistry(testExportDir, SOURCE_DB_EXPORTER_ROLE, testPostgresSource.Source, testPostgresSource.DB(), nil, nil, false)
+	if err != nil {
+		t.Errorf("error initialising name reg for the source: %v", err)
+	}
+	defer testPostgresSource.DB().Disconnect()
+	defer testPostgresSource.ExecuteSqls(cleanUpSqls...)
+	defer testPostgresSource.ExecuteSqls(`DROP TABLE test_source_extra;`)
+
+	metaDB = initMetaDB(testExportDir)
+	//Running the command level functions
+	source = *testPostgresSource.Source
+	source.ExcludeTableList = "test_source_extra"
+
+	expectedPartitionsToRootMap := map[string]string{
+		"public.test_partitions_sequences_l": "public.test_partitions_sequences",
+		"public.test_partitions_sequences_s": "public.test_partitions_sequences",
+		"public.test_partitions_sequences_b": "public.test_partitions_sequences",
+		"p1.london":                          "public.sales_region",
+		"p1.sydney":                          "public.sales_region",
+		"p1.boston":                          "public.sales_region",
+	}
+	expectedTableList := []sqlname.NameTuple{
+		getNameTuple("public.test_partitions_sequences"),
+		getNameTuple("public.test_partitions_sequences_l"),
+		getNameTuple("public.test_partitions_sequences_s"),
+		getNameTuple("public.test_partitions_sequences_b"),
+		getNameTuple("public.sales_region"),
+		getNameTuple("p1.London"),
+		getNameTuple("p1.Sydney"),
+		getNameTuple("p1.Boston"),
+		getNameTuple("public.datatypes1"),
+		getNameTuple("public.foreign_test"),
+	}
+	assertTableListFilteringInTheFirstRun(t, expectedPartitionsToRootMap, expectedTableList)
+
+	//Create msr with required details for subsequent run
+
+	expectedTableListWithOnlyRootTable := []sqlname.NameTuple{
+		getNameTuple("public.test_partitions_sequences"),
+		getNameTuple("public.sales_region"),
+		getNameTuple("public.datatypes1"),
+		getNameTuple("public.foreign_test"),
+	}
+	err = metaDB.UpdateMigrationStatusRecord(func(msr *metadb.MigrationStatusRecord) {
+		msr.SourceDBConf = testPostgresSource.Source
+		msr.TargetDBConf = &testYugabyteDBTarget.Tconf
+		msr.TableListExportedFromSource = lo.Map(expectedTableListWithOnlyRootTable, func(t sqlname.NameTuple, _ int) string {
+			return t.ForOutput()
+		})
+		msr.SourceExportedTableListWithLeafPartitions = lo.Map(expectedTableList, func(t sqlname.NameTuple, _ int) string {
+			return t.ForOutput()
+		})
+		msr.SourceRenameTablesMap = expectedPartitionsToRootMap
+	})
+	if err != nil {
+		t.Fatalf("error updating msr: %v", err)
+	}
+
+	//Start import-data
+	err = testYugabyteDBTarget.Init()
+	if err != nil {
+		utils.ErrExit("Failed to connect to yugabyte database: %w", err)
+	}
+	err = InitNameRegistry(testExportDir, TARGET_DB_IMPORTER_ROLE, nil, nil, &testYugabyteDBTarget.Tconf, testYugabyteDBTarget.TargetDB, false)
+	if err != nil {
+		t.Errorf("error initialising name reg for the source: %v", err)
+	}
+	testYugabyteDBTarget.Finalize()
+	err = testYugabyteDBSource.DB().Connect()
```

Comment:
why do you need to initialize testYugabyteDBSource? 
Can't you run the clean up on testYugabyteDBTarget?

Reply:
`testYugabyteDBSource ` is not just for cleanup, this test case is for `export-data-from-target` which uses source so have to initialize testYugabyteDBSource as well.

Reply:
it looks like this test function only deals with export-data-from-source flow. Just confirm once and remove this initialization if not required.

Reply:
oh yeah, must be a miss in copy-pasting the tests. Removed it.